subtests/docker_cli/events/events.py

```diff
@@ -98,42 +55,88 @@ def is_dupe_event(needle, haystack):
     return False
 
 
+def parse_event_docker_110(line):
+    """
+    Try to parse input as a docker 1.10 event
+    """
+    # eg <timestamp> container start <sha> (details)
+    event_re = re.compile(r'^(?P<timestamp>{timestamp})'
+                          r'\s+(?P<object>\w+)'
+                          r'\s+(?P<operation>{operation})'
+                          r'\s+(?P<identifier>{cid}|{fqin})'
+                          r'\s+\((?P<rest>.*)\)'.format(**regexes))
+    mobj = event_re.match(line)
+    if mobj is None:
+        return None
+
+    # Matched! Extract the positional fields, then try looking for source img
+    details = {
+        'datetime':   DockerTime(mobj.group('timestamp')),
+        'identifier': mobj.group('identifier'),
+        'object':     mobj.group('object'),
+        'operation':  mobj.group('operation'),
+        'source':     None,
+    }
+    # TODO: (maybe): split out components of the parenthesized list.
+    # If so, keep in mind that you can't just split on commas (because
+    # of "Red Hat, Inc.") and that the fields are output in unpredictable
+    # order: even two consecutive event lines will have different ordering.
```

Comment:
Sounds like a bug to me, is it too much to ask for an event-stream that's easily machine-readable?  It's much easier for docker to escape any commas than it is to pull them out on the parsing side.  Don't be afraid to open "design" bugs like this, worst case they get closed but in either case we have to deal with the problem.